ktor-server/ktor-server-plugins/ktor-server-i18n/jvm/src/io/ktor/i18n/I18n.kt

```diff
@@ -62,3 +71,26 @@ public val I18n: ApplicationPlugin<I18nConfiguration> = createApplicationPlugin(
         }
     }
 }
+
+/**
+ * Get the current accepted [locale](java.util.Locale) specified in the HTTP request
+ *
+ * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.i18n.locale)
+ */
+public val ApplicationCall.locale: Locale get() = attributes[LOCALE]
+
+/**
+ * Translate a message key to an accepted language specified in HTTP request
+ *
+ * [Report a problem](https://ktor.io/feedback/?fqname=io.ktor.i18n.i18n)
+ */
+public fun ApplicationCall.i18n(@PropertyKey(resourceBundle = BUNDLE_KEY) key: String): String {
+    val value = attributes[RESOURCE_BUNDLE].getString(key)
+
+    I18N_LOGGER.debug {
+        "translating to $locale: $key=$value"
+    }
+    return value
+}
+
+internal const val BUNDLE_KEY = "messages/messages"
```

Comment:
This value is very uncommon and not recognized by IntelliJ, because Java prefers a class name and not a slash: https://docs.oracle.com/javase/7/docs/api/java/util/ResourceBundle.html#getBundle(java.lang.String,%20java.util.Locale,%20java.lang.ClassLoader,%20java.util.ResourceBundle.Control)

IMHO we should change the key to messages only. While we could make it configurable, IDE support requires a compile time constant.

Reply:
I agree we need to keep the path but did you test it too? I always get errors regarding unresolved property keys 🤔

Reply:
I did some more tests: Turns out, the "new" syntax does not require any files changes.